.github/workflows/codeql-push.yml

```diff
@@ -22,31 +23,30 @@ env:
 jobs:
   CodeQL-Build:
     permissions:
+      actions: read
+      contents: read
       # for github/codeql-action/analyze to upload SARIF results
       security-events: write
       pull-requests: read
     runs-on: ubuntu-20.04
     if: github.repository == 'envoyproxy/envoy'
     steps:
-    - name: Pre-cleanup
-      uses: envoyproxy/toolshed/gh-actions/diskspace@6b3ddd1e42c252d68fb98973760c0ee1943c9c21  # actions-v0.2.20
-      with:
-        to_remove: |
-          /usr/local/lib/android
-
     - name: Checkout repository
       uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1.1
+      with:
+        fetch-depth: 2
 
     - name: Get build targets
       run: |
         # TODO(phlax): Shift this to an action
         compare_head () {
             while IFS= read -r line; do
+                echo "CHECK: ${line}"
```

Comment:
@phlax doesn't this line affect the output of the function to include `CHECK: ...` as well as the output of the bazel query? I am curious since I am now seeing failures from this action in [this PR](https://github.com/envoyproxy/envoy/pull/31090) because of the `CHECK` line. (logs from a failure [here](https://github.com/envoyproxy/envoy/actions/runs/7530448645/job/20496852150?pr=31090))

Reply:
it certainly should be here - altho im a bit confused as it does appear to be working elsewhere - so im thinking its not the cause of your problem - which afaict is more to do with an empty line being returned